src/baseline/expr.rs

```diff
@@ -1029,7 +1029,7 @@ where
                 Intrinsic::ByteNot => self.emit_intrinsic_bin_call(e, dest, intrinsic),
 
                 Intrinsic::BoolEq => self.emit_intrinsic_bin_call(e, dest, intrinsic),
-                Intrinsic::BoolNot => self.emit_intrinsic_bin_call(e, dest, intrinsic),
+                Intrinsic::BoolNot => self.emit_intrinsic_unary_call(e, dest, intrinsic),
```

Comment:
I think it's better to have a separate function for unary intrinsics.

Reply:
Will do!